analysis/descriptive/time_series.R

```diff
@@ -0,0 +1,194 @@
+
+######################################################
+# This script:
+# - Creates four datasets :
+#    1. Prevalence of opioid prescribing in full population;
+#    2. Prevalence of opioid prescribing in people without cancer;
+#    3. Incidence of opioid prescribing in full population;
+#    4. Incidence of opioid prescribing in people without cancer.
+# - each dataset contains monthly time series of both 
+#     any and high dose opioid prescribing, 
+#     broken down by various characteristics
+#######################################################
+
+
+# For running locally only #
+# setwd("C:/Users/aschaffer/OneDrive - Nexus365/Documents/GitHub/opioids-covid-research")
+# getwd()
+
+# Import libraries #
+library('tidyverse')
+library('lubridate')
+library('arrow')
+library('here')
+library('reshape2')
+library('dplyr')
+library('fs')
+library('ggplot2')
+library('RColorBrewer')
+
+## Create directories
+dir_create(here::here("output", "time series"), showWarnings = FALSE, recurse = TRUE)
+dir_create(here::here("output", "joined"), showWarnings = FALSE, recurse = TRUE)
+
+# Read in data
+prev_ts <- read_csv(here::here("output", "joined", "final_timeseries_prev.csv")) %>%
```

Comment:
`read_csv` uses a heuristic to figure out the type of each column based on the first 1000 rows. In this case as the first 1000 rows of age_cat are all `NA` it thinks the column should be logical. To prevent this you should specify the col_types e.g. 
``` 
prev_ts <- read_csv(here::here("output", "joined", "final_timeseries_prev.csv"),
                    col_types = cols(
                      region  = col_character(),
                      imdq10= col_character(),
                      ethnicity  = col_character(),
                      carehome  = col_character(),
                      scd  = col_character(),
                      age_cat  = col_character(),
                      sex = col_character(),
                      date = col_date(format="%Y-%m-%d")
                    ),
              )
```